mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/BehaviorRunner.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming.impl
+
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.actor.typed.{Behavior, Signal}
+
+import scala.collection.immutable.Seq
+
+object BehaviorRunner {
+  sealed trait Interpretable[T]
+
+  final case class StoredMessage[T](message: T) extends Interpretable[T]
+
+  final case class StoredSignal[T](signal: Signal) extends Interpretable[T]
+
+  /**
+   * Interpreter all of the supplied messages or signals, returning
+   * the resulting behavior.
+   */
+  def run[T](behavior: Behavior[T], context: ActorContext[T], stash: Seq[Interpretable[T]]): Behavior[T] =
+    stash.foldLeft(Behavior.start(behavior, context)) {
+      case (b, StoredMessage(msg)) =>
+        val nextBehavior = Behavior.interpretMessage(b, context, msg)
+
+        if ((nextBehavior ne Behaviors.same) && (nextBehavior ne Behaviors.unhandled)) {
+          nextBehavior
+        } else {
+          b
+        }
+
+      case (b, StoredSignal(signal)) =>
+        val nextBehavior = Behavior.interpretSignal(b, context, signal)
+
+        if ((nextBehavior ne Behaviors.same) && (nextBehavior ne Behaviors.unhandled)) {
+          nextBehavior
+        } else {
+          b
+        }
+    }
+}
```

Comment:
Can't you use `StashBuffer` and `unstashAll` which comes out of the box instead?

Reply:
We used to use StashBuffer: https://github.com/akka/alpakka/pull/1853#discussion_r317562543

Reply:
Ah I see, thanks I missed that conversation. That's ok I guess. Could be a future improvement then perhaps. 

The main shortcoming with the home made stash, except for being used/tested by less people, will be that if you use LB telemetry you won't get any metrics for it.